pulp_2_tests/tests/docker/api_v2/test_remove.py

```diff
@@ -74,70 +131,81 @@ def delete_docker_units(self, repo, units):
-        # Verify initial unit count
-        units = self.get_docker_units(self.repo, 'docker_tag')
-        self.assertEqual(len(units), DOCKER_REMOVE['INITIAL']['TAG'], units)
-
-        # Delete by tag - ensure there are no units left
-        self.delete_docker_units(self.repo, units)
-
-        # Count the remaining units
-        remaining_units = self.get_docker_units(self.repo, 'docker_tag')
-        self.assertEqual(len(remaining_units), 0, remaining_units)
-
-    def test_02_remove_manifest_list_all(self):
-        """Sync docker repo and remove all manifest_lists."""
-        # Verify initial unit count
-        units = self.get_docker_units(self.repo, 'docker_manifest_list')
-        self.assertEqual(len(units), DOCKER_REMOVE['INITIAL']['MANIFEST_LIST'], units)
-
-        # Delete by tag - ensure there are no units left
-        self.delete_docker_units(self.repo, units)
-
-        # Count the remaining units
-        remaining_units = self.get_docker_units(self.repo, 'docker_manifest_list')
-        self.assertEqual(len(remaining_units), 0, remaining_units)
-
-    def test_03_remove_manifest_all(self):
-        """Sync docker repo and remove all manifests."""
-        # Verify initial unit count
-        units = self.get_docker_units(self.repo, 'docker_manifest')
-        self.assertEqual(len(units), DOCKER_REMOVE['INITIAL']['MANIFEST'], units)
-
-        # Delete by tag - ensure there are no units left
-        self.delete_docker_units(self.repo, units)
-
-        # Count the remaining units
-        remaining_units = self.get_docker_units(self.repo, 'docker_manifest')
-        self.assertEqual(len(remaining_units), 0, remaining_units)
-
-    # pylint: disable=R0201
-    def test_04_remove_manifest_list_not_shared(self):
-        """Sync docker repo and remove some non-shared manifest_lists."""
-        raise unittest.SkipTest('Stubbed test case, Not Implemented Yet')
-
-    def test_05_remove_manifest_not_shared(self):
-        """Sync docker repo and remove some non-shared manifests."""
-        raise unittest.SkipTest('Stubbed test case, Not Implemented Yet')
-
-    def test_06_remove_manifest_list_shared(self):
-        """Sync docker repo and remove some shared manifest_lists."""
-        raise unittest.SkipTest('Stubbed test case, Not Implemented Yet')
-
-    def test_07_remove_manifest_shared(self):
-        """Sync docker repo and remove some shared manifests."""
-        raise unittest.SkipTest('Stubbed test case, Not Implemented Yet')
-
-    def test_08_sync_remove_units(self):
-        """Repo A. Sync to fixture-1. Sync to fixture-2."""
-        raise unittest.SkipTest('Stubbed test case, Not Implemented Yet')
+    def test_00_test(self):
+        """foo"""
+        repo = self.create_and_copy_test_repo(self.repo, NON_SHARED_MANIFEST_LIST)
+        self.addCleanup(self.client.delete, repo['_href'])
+
+        docker_units = self.get_docker_units(repo, DOCKER_UNIT_TYPES)
```

Comment:
docker_untis will be a dict. Perhaps we can use docker_units.items(): then parse the key, value.

Reply:
Yeah, I looked that up right before the meeting. Part of the optimization/learning i need to do on iterating through dicts.